cp3pt0-deployment/common/utils.sh

```diff
@@ -231,6 +231,67 @@ function wait_for_operand_request() {
     wait_for_condition "${condition}" ${retries} ${sleep_time} "${wait_message}" "${success_message}" "${error_message}"
 }
 
+function wait_for_nss_patch() {
+    local namespace=$1
+    local csv_name=$($OC get sub ibm-common-service-operator -n ${namespace} -o jsonpath='{.status.installedCSV}')
+    local condition="${OC} -n ${namespace}  get csv ${csv_name} -o jsonpath='{.spec.install.spec.deployments[0].spec.template.spec.containers[0].env[?(@.name==\"WATCH_NAMESPACE\")].valueFrom.configMapKeyRef.name}'| grep 'namespace-scope'"
+    local retries=10
+    local sleep_time=10
+    local total_time_mins=$(( sleep_time * retries / 60))
+    local wait_message="Waiting for cs-operator CSV to be patched with NSS configmap"
+    local success_message="cs-operator CSV is patched with NSS configmap"
+    local error_message="Timeout after ${total_time_mins} minutes waiting for cs-operator CSV to be patched with NSS configmap"
+    local pod_name=$($OC get pod -n ${namespace} | grep namespace-scope | awk '{print $1}')
+
+    # wait for nss patch
+    info "${wait_message}"
+    while true; do
+        result=$(eval "${condition}")
+
+        # restart namespace scope operator pod to reconcilie
+        if [[ ( ${retries} -eq 0 ) && ( ! -z "${result}" ) ]]; then
+            info "Reconciling namespace scope operator"
+            echo "deleting pod ${pod_name}"
+            $OC delete pod ${pod_name} -n ${namespace}    
+            wait_for_condition "${condition}" ${retries} ${sleep_time} "${wait_message}" "${success_message}" "${error_message}"
+            break
+        fi
+ 
+        sleep ${sleep_time}
+        result=$(eval "${condition}")
+        
+        if [[ ! -z "${result}" ]]; then
+            info "RETRYING: ${wait_message} (${retries} left)"
+            retries=$(( retries - 1 ))
+        else
+            break
+        fi
+    done
+
+    if [[ ! -z "${success_message}" ]]; then
+        success "${success_message}"
+    fi
+    
+    # wait for deployment to be ready
+    deployment_name=$($OC get deployment -n ${namespace} | grep common-service-operator | awk '{print $1}')
+    wait_for_deployment ${namespace} ${deployment_name}
+    
+}
+
+function wait_for_deployment() {
+    local namespace=$1
+    local name=$2
+    local condition="${OC} -n ${namespace} get deployment ${name} --no-headers --ignore-not-found -o jsonpath='{.status.readyReplicas}' | grep '1'"
+    local retries=10
+    local sleep_time=30
+    local total_time_mins=$(( sleep_time * retries / 60))
+    local wait_message="Waiting for Deployment ${name} to be ready"
+    local success_message="Deployment ${name} is running"
+    local error_message="Timeout after ${total_time_mins} minutes waiting for Deployment ${name} to be running"
+    sleep 30
```

Comment:
```suggestion
```

Unnecessary. Why wait before the actual wait function?

Reply:
The previous function is to check the WATCH_NAMESPACE in CSV, and this function is to check if the deployment is ready. I think OLM could use some time to modify the deployment, so I want to give OLM 30 seconds to apply the changes in CSV to the deployment.

Reply:
Instead of waiting for deployment ready, make the condition check the deployment's environment variables for existence of namespace-scope, similar to how it checks CSV for it https://github.com/IBM/ibm-common-service-operator/pull/1077/files#diff-838030827130a186b5a29a614513901c9628da0149119e7357e4a7cb67e5552cR237.

This kind of check is more accurate, and will cover edge case where OLM might be having issues, blocking it from updating deployment. This edge case would cause the current deployment ready check to pass when it should not.